### DISCUSSION TITLE
Any tools to render side-by-side annotations from source code?

### DISCUSSION THREAD

**Discussion by miaotianyi** [labmlai/annotated_deep_learning_paper_implementations#181] [General] [answered]
I'm wondering if there's an open source tool to generate a website like https://nn.labml.ai/ (side-by-side, line-by-line rendering of code and annotations) from source code directly. If so, what are some of the formats that the python comments should follow? Is there a service like readthedocs that can host websites like this?

**Comment by hnipun** [comment, marked as answer]
If you want to try out https://lit.labml.ai, you'll find that its format is easy to follow. 

As an example, take a look at https://github.com/vpj/rl_samples/blob/master/ppo.py which renders https://lit.labml.ai/github/vpj/rl_samples/blob/master/ppo.py.

**Reply by yong3yuan3yong4** [reply]
Thanks! I attempted to alter the .py files, but this didn't result in any changes to the corresponding .html files. 

It seems that modifications to the .html files are only recognized when they are made directly within the .html file itself. 

But the code within the .html file is quite  messy, making it difficult to work with. Additionally, I was unable to locate any explicit code to link to the .py files within the .html file.

**Reply by gziz** [reply]
Any updates?

**Reply by fengweipeng** [reply]
https://lit.labml.ai/ is not available.
